lnrpc/switchrpc/switch_server.go

```diff
@@ -86,12 +88,43 @@ type ServerShell struct {
 type Server struct {
 	cfg *Config
 
+	// usedAttemptIDs maintains the set of attempt IDs that have been used
+	// in either SendOnion or TrackOnion. This ensures that if TrackOnion
+	// returns PAYMENT_ID_NOT_FOUND or SendOnion initiates HTLC creation for
+	// a given attempt ID, SendOnion cannot subsequently succeed with the
+	// same attempt ID. This mechanism safeguards against overpayment in
+	// scenarios where network requests are reordered. If an attempt ID has
+	// already been used by either SendOnion or TrackOnion, SendOnion will
+	// return DUPLICATE_HTLC for that attempt ID.
+	usedAttemptIDs *roaring64.Bitmap
```

Comment:
as noted offline, an in-mem solution is not enough to make something idempotent. Will need a persisted solution if we find that we indeed are at rish of duplicate attempts